frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.test.tsx

```diff
@@ -934,4 +1032,171 @@ describe('DistributionField hypershift clusters', () => {
+      owner: {},
+      isHypershift: true,
+      hypershift: {
+        agent: false,
+        secretNames: [],
+        hostingNamespace: '',
+        isUpgrading: false,
+      },
+      isRegionalHubCluster: false,
+    }
+
+    const { queryByRole } = await renderDistributionInfoField(
+      mockHypershiftCluster,
+      true,
+      false,
+      undefined,
+      undefined,
+      mockHostedCluster,
+      false
+    )
+
+    expect(queryByRole('progressbar')).toBeFalsy()
+  })
+
+  it("Shouldn't show HCP upgrading, upgrading but on nodepools table", async () => {
+    const { queryAllByText, queryByRole } = await renderDistributionInfoField(
+      mockHypershiftCluster,
+      true,
+      false,
+      undefined,
+      undefined,
+      mockHostedCluster,
+      false,
+      'nodepool'
+    )
+
+    expect(queryAllByText(/upgrading to 4\.11\.22/i).length).toBe(0)
+    expect(queryByRole('progressbar')).toBeFalsy()
+  })
+
+  it('Should show upgrading but with unavailable version num', async () => {
+    const mockHostedCluster: HostedClusterK8sResource = {
+      apiVersion: HostedClusterApiVersion,
+      kind: HostedClusterKind,
+      spec: {
+        dns: {
+          baseDomain: 'dev06.red-chesterfield.com',
+        },
+        release: {
+          image: 'randomimage',
+        },
+        services: [],
+        platform: {},
+        pullSecret: { name: 'psecret' },
+        sshKey: { name: 'thekey' },
+      },
+      status: {
+        conditions: [
+          {
+            lastTransitionTime: '2022-12-17T22:14:15Z',
+            message: 'The hosted control plane is available',
+            observedGeneration: 4,
+            reason: 'HostedClusterAsExpected',
+            status: 'True',
+            type: 'Available',
+          },
+        ],
+        version: {
+          desired: {
+            image: '',
+          },
+          history: [
+            {
+              completionTime: '',
+              image: '',
+              startedTime: '2022-10-24T20:34:08Z',
+              state: 'Partial',
+              verified: false,
+              version: '',
+            },
+          ],
+          observedGeneration: 2,
+        },
+      },
+    }
+    const { queryAllByText, queryByRole } = await renderDistributionInfoField(
+      mockHypershiftCluster,
+      true,
+      false,
+      undefined,
+      undefined,
+      mockHostedCluster,
+      false
+    )
+
+    expect(queryAllByText(/upgrading cluster/i).length).toBe(1)
+    expect(queryByRole('progressbar')).toBeTruthy()
+  })
+
+  it('Should have cluster name and version in popover', async () => {
+    const { getByText, queryAllByText, queryByRole } = await renderDistributionInfoField(
+      mockHypershiftCluster,
+      true,
+      false,
+      undefined,
+      undefined,
+      mockHostedCluster,
+      false
+    )
+
+    //await new Promise((resolve) => setTimeout(resolve, 500)) // makes sure everything is finished
```

Comment:
I think you forgot to remove the comment and debug line.

Reply:
Removed